[ADVAPP-568]: Add light documentation on new permission creation (#811)
canyongbs/advisingapp: docs/data-migrations.md

When creating data migrations it is important to adhere to the following rules:

1. Do **NOT** use references to classes and facades that could be removed in the future. This includes Eloquent models. Currently, the only accepted classes to use within migrations are as follows:
    - `Illuminate\Support\Facades\DB`
        - Used to Interact with the database.
    - `Laravel\Pennant\Feature`
        - To interact with feature flagging.

> 2. As mentioned above, ensure all possible SQL errors are captured and handled. (UniqueConstraintExeception for example) Also that any database change is surrounded by a transaction to prevent corruption of the query connection.

Reviewer: ```suggestion
2. As mentioned above, ensure all possible SQL errors are captured and handled, such as `UniqueConstraintExeception`. Also ensure that any database change is surrounded by a transaction to prevent corruption of the query connection.
```

> 2. Ensure that the migration is idempotent. This means that the migration can be run multiple times without causing any issues. Tables and columns should be checked for the existence or SQL errors are caught and properly handled, data checked if it is in the expected state before it is changed, etc.

Reviewer: ```suggestion
2. Ensure that the migration is idempotent. This means that the migration can be run multiple times without causing any issues. Tables and columns should be checked for the existence, SQL errors should be caught and properly handled, and data should be checked if it is in the expected state before it is changed, etc.
```

3. You **MUST** include a `down` method in the migration.
    - In production, Data migrations are intended to be run once and not rolled back. If you need to fix a data migration that has been run in Production already, you should create a new data migration to do so.
    - The reason we add a `down` to our data migrations is to facilitate testing of these data migrations in special cases.
4. Do **NOT** include schema changes in data migrations. Schema changes should be done in schema migrations. Data migrations should only be used to manipulate data.

> 5. If you are adding permissions to the application, that technically is a data migration. But we consider it a special type. Details on how to do so can be found in the [Roles and Permisisons](/docs/roles-and-permissions.md) documentation.

Reviewer: ```suggestion
5. If you are adding permissions to the application, that is a special type of data migration which can be read about in the [Roles and Permisisons](/docs/roles-and-permissions.md) documentation.
```
